### DISCUSSION TITLE
Mannual vs Automatic Vectorization

### DISCUSSION THREAD

**Discussion by EXing** [facebookresearch/theseus#594] [Q&A] [answered]
## ❓ Questions and Help

#503 said 

> Note that this feature is mostly for programming convenience. The alternative of creating individual cost functions objects doesn't hurt performance, because TheseusLayer by defaults vectorizes cost functions with the same pattern so they are evaluated in a single call.

But #565 shows manual vectorization makes the optimization faster.

This makes me confused.

**Comment by luisenp** [comment, marked as answer]
The sentence is #503 is indeed misleading. There is some overhead associated with cost function vectorization, because the populating the data is done with for loops in Python. So if you have a really high number of costs being vectorized together, you may pay a significant cost.
